lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -838,80 +801,190 @@ spec = do
                 [ expectField #delegation (`shouldBe` notDelegating [])
                 ]
 
-    it "STAKE_POOLS_JOIN/QUIT - Checking delegation expectations with \
-       \multiple certs inserted in a given epoch" $ \ctx -> do
-        (_, p1:p2:p3:_) <- eventually "Stake pools are listed" $
+    it "STAKE_POOL_NEXT_02 - Override join with join in the same epoch =>\
+        \ delegating to the last one in the end" $ \ctx -> do
+        (_, p1:p2:_) <- eventually "Stake pools are listed" $
             unsafeRequest @[ApiStakePool] ctx Link.listStakePools Empty
         w <- fixtureWallet ctx
 
+        request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty
+            >>= flip verify
+                [ expectField #delegation (`shouldBe` notDelegating []) ]
+
+        -- make sure we are at the beginning of new epoch
+        (currentEpoch, sp) <- getSlotParams ctx
+        waitForNextEpoch ctx
+
+        let joinPool pool = do
```

Comment:
maybe add another arg for delegation expected field and use throughout all added code?